src/main/webapp/meal-page-script.js

```diff
@@ -13,7 +13,13 @@
 // limitations under the License.
 
 function fetchMealInfo() {
-    fetch('/meal').then(response => response.json()).then((meal) => 
+    // use mapping /meal.html?id=<id> 
+    // fetches form server by action meal/<id>
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const id = urlParams.get("id");
+
+    fetch('/meal/' + id.toString()).then(response => response.json()).then((meal) => 
```

Comment:
Once you get a response, you should check whether `response.ok` is true.

If the backend responds with some kind of error (such as "not found"), then the response will still be there. However, the status of the response will be set to an error code (usually 4XX). To check that quickly, the `Response` interface has an `ok` property that is true only if the status code points to success (that is: 2XX).

When `fetch` gets a "404 not found" response, `.json()` may or may not succeed - there may be a body to the error response (such as an error message), and in both cases the consequences may be confusing. To avoid that, please check the response status.

Reply:
Yep.
One thing that may be a bit counterintuitive in the fetch api is that it will only reject the promise on a network error, not a http error [link](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch). Another words, the `fetch` promise will get rejected if you lose your internet connection, but not if you point it to a non-existent url.
Handling non-ok statuses has to be done manually by inspecting the `ok` field. You can find some examples on the MDN page, or you can google something along the lines of "fetch http error handling".